src/model/Board.java

```diff
@@ -48,6 +48,25 @@ private void decrementPieceCount(int position) {
         }
     }
 
+    public PieceColor determineWinner() {
+        final boolean whiteHasNoPieces = this.getNumberOfWhitePieces() == 0;
+        final boolean whiteHasNoMoves = MoveGenerator.getAllPossibleMoves(this, PieceColor.WHITE)
+                .isEmpty();
+        final boolean blackHasNoPieces = this.getNumberOfBlackPieces() == 0;
+        final boolean blackHasNoMoves = MoveGenerator.getAllPossibleMoves(this, PieceColor.BLACK)
+                .isEmpty();
+
+        final boolean whiteWins = blackHasNoPieces || blackHasNoMoves;
+        final boolean blackWins = whiteHasNoPieces || whiteHasNoMoves;
+
+        if (whiteWins) {
+            return PieceColor.WHITE;
+        } else if (blackWins) {
+            return PieceColor.BLACK;
+        }
+        return null;
```

Comment:
How about we just remove the determineWinner() method entirely and change playGame() to something like the following:
```
private void playGame() {
        this.choosePlayerTypes();

        PieceColor currentColor = PieceColor.BLACK;
        Player currentPlayer = this.blackPlayer;

        System.out.println("Game start");
        CommandLineHelper.printBoard(this.gameBoard);
        while (!this.gameBoard.isEndState(currentColor)) {
            currentPlayer.makeMove(this.gameBoard);
            CommandLineHelper.printBoard(this.gameBoard);
            currentColor = currentColor.getOppositeColor();
            currentPlayer = this.getOtherPlayer(currentPlayer);

            try {
                Thread.sleep(500);
            } catch (final InterruptedException e) {
                e.printStackTrace();
            }
        }
        this.displayWinner(currentColor.getOpposite());
    }
```